arangod/Graph/Providers/ClusterProvider.cpp

```diff
@@ -105,17 +105,19 @@ ClusterProvider<StepImpl>::ClusterProvider(
 
 template<class StepImpl>
 ClusterProvider<StepImpl>::~ClusterProvider() {
-  clear();
+  clear(true);  // Make sure we actually free all memory in the edge cache!
```

Comment:
would be more readable if we would have two functions `force_clear()` and `try_clear()` (or similar) instead of a boolean flag.

Reply:
Changed. We do not want to change the name `clear` since it is used everywhere. But `clear(true)` can easily be replaced by `clearWithForce()`. Done.